source/common/common/utility.h

```diff
@@ -551,6 +552,43 @@ template <class Value> struct TrieEntry {
  * A trie used for faster lookup with lookup time at most equal to the size of the key.
  */
 template <class Value> struct TrieLookupTable {
+  /**
+   * A custom destructor helps avoid a stack overflow issue when trie depth is great and default
+   * destructor recursively collects unique_ptr.
+   */
+  ~TrieLookupTable() {
+    // Use a stack to store non-nullptr nodes which should be deleted, and their children will be
+    // pushed into this stack and will be deleted later.
+    std::stack<TrieEntry<Value>*> to_delete;
+    // Release all pointers in root_ and push them into the stack.
+    moveChildTriesToStack(&root_, to_delete);
+    // Now all unique_ptr in root_ are released and raw pointers are stored inside the stack.
+    // We will delete all of the raw pointers in below while loop.
+    // All raw pointers in to_delete will finally be deleted and popped out.
+    while (!to_delete.empty()) {
+      TrieEntry<Value>* current = to_delete.top();
+      to_delete.pop();
+      moveChildTriesToStack(current, to_delete);
+      // Now all the entries inside current node are released and pushed into the stack.
+      // We can safely delete it now.
```

Comment:
nit: this comment's great, suggest consolidating with the comment at line 567?

Reply:
> consolidating

Modified the comments. Does it look better now?